packages/opentelemetry-core/src/context/context.ts

```diff
@@ -84,3 +91,34 @@ export function getParentSpanContext(
 ): SpanContext | undefined {
   return getActiveSpan(context)?.context() || getExtractedSpanContext(context);
 }
+
+/**
+ * Sets value on context to indicate that instrumentation should
+ * be suppressed beyond this current scope.
+ *
+ * @param context context to set the suppress instrumentation value on.
+ */
+export function suppressInstrumentation(context: Context): Context {
+  return context.setValue(SUPPRESS_INSTRUMENTATION_KEY, true);
+}
+
+/**
+ * Sets value on context to indicate that instrumentation should
+ * no-longer be suppressed beyond this current scope.
+ *
+ * @param context context to set the suppress instrumentation value on.
+ */
+export function unsuppressInstrumentation(context: Context): Context {
+  return context.setValue(SUPPRESS_INSTRUMENTATION_KEY, false);
+}
+
+/**
+ * Return current suppress instrumentation value for the given context,
+ * if it exists.
+ *
+ * @param context context check for the suppress instrumentation value.
+ */
+export function isInstrumentationSuppressed(context: Context): boolean {
+  const value = context.getValue(SUPPRESS_INSTRUMENTATION_KEY) as boolean;
+  return !!value;
```

Comment:
I would remove the extraneous cast and assignment, but this is minor.

```suggestion
  return Boolean(context.getValue(SUPPRESS_INSTRUMENTATION_KEY));
```

Reply:
I was definitely giving that the side-eye while I was restructuring. Thanks for the extra push. :)

Reply:
It's probably minor, but since this function is likely to be called _a lot_, I think the performance benefit alone of skipping the assignment is worth it since you don't really lose anything here.